### ISSUE TITLE
A bug in FP16 training of VAE

### ISSUE THREAD

**Issue by serend1p1ty** [PKU-YuanGroup/Open-Sora-Plan#335]
In fact, the problem is caused by PyTorch Lightning, and **may decrease the performance of VAE**.

PyTorch Lightning has recognized this issue and list it in 2.3.x milestones.
https://github.com/Lightning-AI/pytorch-lightning/issues/18089
https://github.com/Lightning-AI/pytorch-lightning/milestone/50

# Problem Introduction

We do manual optimization in `CausalVAEModel._training_step_gan()`.

https://github.com/PKU-YuanGroup/Open-Sora-Plan/blob/b08681f697658c81361e1ec6c07fba55c79bb4bd/opensora/models/ae/videobase/causal_vae/modeling_causalvae.py#L462

```
opt1.zero_grad()
self.manual_backward(aeloss)
self.clip_gradients(opt1, gradient_clip_val=1, gradient_clip_algorithm="norm")
opt1.step()
```

If we enable mixed precision, `self.manual_backward(aeloss)` will scale the loss before backward it, i.e., `grad_scaler.scale(loss).backward()`.

Therefore, **before gradient clipping, we need to unscale the gradient**. But the current code lacks this step.

The ideal code should be like

```
opt1.zero_grad()
self.manual_backward(aeloss)
self.unscale_gradients()
self.clip_gradients(opt1, gradient_clip_val=1, gradient_clip_algorithm="norm")
opt1.step()
```

# How to Fix

I think we should give up PyTorch Lightning and write a simple training loop as as an alternative.

**Comment by qqingzheng** [comment]
Thank you, we have abandoned pytorch ligjtning training internally. New VAE training code will be released in the future.
